chia/consensus/multiprocess_validation.py

```diff
@@ -274,34 +263,25 @@ async def pre_validate_blocks_multiprocessing(
         except ValueError:
             return [PreValidationResult(uint16(Err.INVALID_SUB_EPOCH_SUMMARY.value), None, None, False, uint32(0))]
 
-        if block_rec.sub_epoch_summary_included is not None:
-            if wp_summaries is not None:
-                idx = int(block.height / constants.SUB_EPOCH_BLOCKS) - 1
-                next_ses = wp_summaries[idx]
-                if not block_rec.sub_epoch_summary_included.get_hash() == next_ses.get_hash():
-                    log.error("sub_epoch_summary does not match wp sub_epoch_summary list")
-                    return [
-                        PreValidationResult(uint16(Err.INVALID_SUB_EPOCH_SUMMARY.value), None, None, False, uint32(0))
-                    ]
+        if block_rec.sub_epoch_summary_included is not None and wp_summaries is not None:
+            idx = int(block.height / constants.SUB_EPOCH_BLOCKS) - 1
+            next_ses = wp_summaries[idx]
+            if not block_rec.sub_epoch_summary_included.get_hash() == next_ses.get_hash():
+                log.error("sub_epoch_summary does not match wp sub_epoch_summary list")
+                return [PreValidationResult(uint16(Err.INVALID_SUB_EPOCH_SUMMARY.value), None, None, False, uint32(0))]
+
         # Makes sure to not override the valid blocks already in block_records
-        if not block_records.contains_block(block_rec.header_hash):
-            block_records.add_block_record(block_rec)  # Temporarily add block to dict
+        if not blockchain.contains_block(block_rec.header_hash):
```

Comment:
i think this can be a check inside augmented_blockchain, would make more sense and simplify this code

Reply:
originally my intention was to remove this check entirely, and just make sure `AugmentedBlockchain` always checks the underlying blockchain first, to get the same effect. I can't really imagine a scenario where it makes a difference though. But with a long-lived `AugmentedBlockchain` object, I can imaging blocks being added to the underlying chain and maybe are left in the extra blocks for a while. I'll experiment a bit more to see if I can seimplify this further.

Reply:
but AugmentedBlockchain is only used on individual batches in what scenario would it be long lived and why would we want that ?

Reply:
when we pipeline validation and `add_block()`. Say we validate the first batch, we then pass that batch on to the `add_block()` pipeline step and we move on to validate the second batch. The first batch hasn't been added to the blockchain yet, so we still need to use the `AugmentedBlockchain` that pretends that the first batch was added.